Formula/synergy-core.rb

```diff
@@ -0,0 +1,121 @@
+class SynergyCore < Formula
+  desc "Synergy, the keyboard and mouse sharing tool"
+  homepage "https://symless.com/synergy"
+
+  # The synergy-core/LICENSE file contains the following preamble:
+  #    This program is released under the GPL with the additional exemption
+  #    that compiling, linking, and/or using OpenSSL is allowed.
+  # This preamble is followed by the text of the GPL-2.0.
+  #
+  # The SPDX license exception list does not contain a symbol corresponding
+  # to synergy-core's OpenSSL exception. The openvpn-openssl-exception
+  # exception symbol is very similar but it is not exactly the same.
+  #
+  # Synergy is free and open source software, so it can be included in brew.
+  # However, the brew `license` method does not provide a way to express the
+  # fact that Synergy is released under "GPL-2.0-only" plus an exception.
+  # Therefore, the following `license` declaration is technically inaccurate.
+  license "GPL-2.0-only"
+
+  head "https://github.com/symless/synergy-core.git", branch: "master"
+
+  stable do
+    url "https://github.com/symless/synergy-core/archive/refs/tags/1.14.3.5-stable.tar.gz"
+    sha256 "9259fc8aa02610121932cd5887c4d878beb08b2f7190e834c0584d53b36f5f9b"
+
+    # Use the version of pugixml from brew, instead of the version included in
+    # a submodule of the synergy-core repository.
+    #
+    # Upstream has merged the following patch, but it is not included in a
+    # stable release yet. See:
+    #   https://github.com/symless/synergy-core/commit/35d7d2754d98d299f24ccb3c
+    #
+    # This `patch` block can be removed once synergy-core publishes a new stable
+    # release that includes the patch.
+    patch do
+      url "https://github.com/symless/synergy-core/commit/35d7d2754d98d299f24ccb3cb294538a58d83d5d.patch?full_index=1"
+      sha256 "7db4a7689f02b5bfc9bdf2c6aa015cb174e17eb0bc7ee412d20e5c286f53a47a"
+    end
+  end
+
+  depends_on "cmake" => :build
+  depends_on "openssl@3"
+  depends_on "pugixml"
+  depends_on "qt@5"
+
+  on_linux do
+    depends_on "gcc" => :build # For C++17 support.
```

Comment:
```suggestion
    depends_on "gcc" # For C++17 support.
```
You probably have linkage with `libstdc++` so you'll still need this at runtime.